Source/JavaScriptCore/lol/LOLJIT.cpp

```diff
@@ -0,0 +1,1612 @@
+        DEFINE_OP(op_profile_type)
+        DEFINE_OP(op_profile_control_flow)
+        DEFINE_OP(op_get_parent_scope)
+        DEFINE_OP(op_put_by_id)
+        DEFINE_OP(op_put_by_val_direct)
+        DEFINE_OP(op_put_by_val)
+        DEFINE_OP(op_put_private_name)
+        DEFINE_OP(op_put_getter_by_id)
+        DEFINE_OP(op_put_setter_by_id)
+        DEFINE_OP(op_put_getter_setter_by_id)
+        DEFINE_OP(op_put_getter_by_val)
+        DEFINE_OP(op_put_setter_by_val)
+        DEFINE_OP(op_to_property_key)
+        DEFINE_OP(op_to_property_key_or_number)
+
+        DEFINE_OP(op_get_internal_field)
+        DEFINE_OP(op_put_internal_field)
+
+        DEFINE_OP(op_iterator_open)
+        DEFINE_OP(op_iterator_next)
+
+        DEFINE_OP(op_ret)
+        DEFINE_OP(op_rshift)
+        DEFINE_OP(op_unsigned)
+        DEFINE_OP(op_urshift)
+        DEFINE_OP(op_set_function_name)
+        DEFINE_OP(op_stricteq)
+        DEFINE_OP(op_sub)
+        DEFINE_OP(op_switch_char)
+        DEFINE_OP(op_switch_imm)
+        DEFINE_OP(op_switch_string)
+        DEFINE_OP(op_throw)
+        DEFINE_OP(op_to_number)
+        DEFINE_OP(op_to_numeric)
+        DEFINE_OP(op_to_string)
+        DEFINE_OP(op_to_object)
+        DEFINE_OP(op_to_primitive)
+
+        DEFINE_OP(op_resolve_scope)
+        DEFINE_OP(op_get_from_scope)
+        DEFINE_OP(op_put_to_scope)
+        DEFINE_OP(op_get_from_arguments)
+        DEFINE_OP(op_put_to_arguments)
+
+        DEFINE_OP(op_log_shadow_chicken_prologue)
+        DEFINE_OP(op_log_shadow_chicken_tail)
+
+        default:
+            RELEASE_ASSERT_NOT_REACHED();
+        }
+        if (sizeMarker) [[unlikely]]
+            m_vm->jitSizeStatistics->markEnd(WTFMove(*sizeMarker), *this, Ref(m_plan));
+
+        if (LOLJITInternal::verbose)
+            dataLogLn("At ", bytecodeOffset, ": added ", m_slowCases.size() - previousSlowCasesSize, "(", m_slowCases.size(), ") allocator: ", m_fastAllocator);
+    }
+
+#undef NEXT_OPCODE_IN_MAIN
+#undef DEFINE_SLOW_OP
+#undef DEFINE_OP
+}
+
+void LOLJIT::privateCompileSlowCases()
+{
+#define DEFINE_SLOWCASE_OP(name) \
+    case name: { \
+        if (!isImplemented(name)) \
+            m_replayAllocator.flushAllRegisters(*this); \
+        emitSlow_##name(currentInstruction, iter); \
+        break; \
+    }
+
+#define DEFINE_SLOWCASE_SLOW_OP(name, Struct) \
+    case op_##name: { \
+        if constexpr (isImplemented(op_##name)) { \
+            emitCommonSlowPathSlowCaseCall<Struct>(currentInstruction, iter, slow_path_##name); \
+        } else { \
+            m_replayAllocator.flushAllRegisters(*this); \
+            emitSlowCaseCall(iter, slow_path_##name); \
+        } \
+        break; \
+    }
+
+#define REPLAY_ALLOCATION_FOR_OP(name, Struct) \
+    case name: { \
+        ASSERT(isImplemented(name)); \
+        m_replayAllocator.allocate(*this, currentInstruction->as<Struct>(), m_bytecodeIndex); \
+        break; \
+    }
+
+    // m_replayAllocator.reset();
```

Comment:
Is this expected?

Reply:
Ah, that was left over from an earlier iteration where replay and fast allocator were the same. Will remove.